File: inst/materializer/js/materialize-like.js
$(document).ready(function () {
      $(".materialize-like").on("click", function () {
         var el = $(this).find("i");
        if (el.text() === "thumb_up") {
           el.text("thumb_down");
           el.removeClass("grey-text text-lighten-2");
           el.removeClass("green-text text-lighten-2");
           el.addClass("red-text text-lighten-2");
         }
         else if (el.text() === "thumb_down") {
            el.text("thumbs_up_down");
            el.removeClass("green-text text-lighten-2");
            el.removeClass("red-text text-lighten-2");
            el.addClass("grey-text text-lighten-2");
         }
         else if (el.text() === "thumbs_up_down") {
            el.text("thumb_up");
            el.removeClass("red-text text-lighten-2");
            el.removeClass("grey-text text-lighten-2");
            el.addClass("green-text text-lighten-2");
         }
         $(this).change();
      });

     var materializeLike = new Shiny.InputBinding();
     $.extend(materializeLike, {
         find: function (scope) {
             return $(scope).find(".materialize-like");
         },
         getValue: function (el) {
            var value;
            switch ($(el).find("i").text()) {
              case "thumb_up":
                value = true;
                break;
              case "thumb_down":
                value = false;
                break;
              case "thumbs_up_down":
                value = null;
            }
            return value;
         },
         subscribe: function (el, callback) {
             $(el).on("change.materialize-like", function (e) {
                 callback();
             });
         },
         unsubscribe: function (el) {
             $(el).off(".materialize-like");
         },
         receiveMessage: function(el, data) {
            if (data.disabled !== undefined) {
               $(el).css("pointer-events", data.disabled ? "none" : "auto");
            }

            var $el = $(el).find("i");
            if (data.liked === undefined) {
               $el.text("thumbs_up_down");
               $el.removeClass("red-text text-lighten-2");
               $el.removeClass("green-text text-lighten-2");
               $el.addClass("grey-text text-lighten-2");
            }
            else if (data.liked === true) {
               $el.text("thumb_up");
               $el.removeClass("red-text text-lighten-2");
               $el.removeClass("grey-text text-lighten-2");
               $el.addClass("green-text text-lighten-2");
            }
            else if (data.liked === false) {
               $el.text("thumb_down");
               $el.removeClass("grey-text text-lighten-2");
               $el.removeClass("green-text text-lighten-2");
               $el.addClass("red-text text-lighten-2");
            }
            $(el).change();
         }

      });

   Shiny.inputBindings.register(materializeLike, "materialize.like");
 });

File: R/materialize-like.R
#' Create a materialize like element
#'
#' Build a materialize like and dislike element
#' @param inputId String. The input identifier used to access the value.
#' @param liked Boolian. The initial status of the element. Could be TRUE, FALSE or NULL
#' @param size String. Size of icon, could be 'tiny' (1rem), 'small' (2rem), 'medium' (4rem) and 'large' (6rem).
#' @param disabled Boolean. Disable iteractivity (click).
#' @examples
#' require(materializer)
#' material_like(
#'   inputId = "example_checkbox",
#'   liked = TRUE
#' )
#' @export
material_like <- function(inputId, liked = NULL, size = "medium", disabled = FALSE) {

  shiny::tags$span(
    id = inputId,
    class = paste("materialize-like"),
    style = paste0("pointer-events:",ifelse(disabled,"none","auto")),
    material_icon(
      ifelse(is.null(liked),"thumbs_up_down",ifelse(liked, "thumb_up", "thumb_down")),
      align = "center",
      size = size,
      color = ifelse(is.null(liked),"grey lighten-2",ifelse(liked, "green lighten-2", "red lighten-2"))
    ),
    includeInHead(
      "materialize-like.js"
    )
  )
}


#' @rdname material_like
#' @param session Shiny default reactive domain.
#' @export
update_material_like <- function(inputId, liked = NULL, disabled = NULL, session = shiny::getDefaultReactiveDomain()) {
  session$sendInputMessage(
    inputId,
    cleanList(
      liked = liked,
      disabled = disabled
    )
  )
}
